### PR TITLE
Remove inefficient use of std::string in C++ transformation plugins.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

**Comment by jpeach** [comment]
How is `string_view` more performant than a `const string&`?

File: lib/cppapi/GzipDeflateTransformation.cc [hunk 1]
```diff
@@ -94,7 +93,7 @@ GzipDeflateTransformation::consume(const string &data)
 
   int iteration               = 0;
   state_->z_stream_.data_type = Z_ASCII;
-  state_->z_stream_.next_in   = reinterpret_cast<unsigned char *>(const_cast<char *>(data.c_str()));
+  state_->z_stream_.next_in   = reinterpret_cast<unsigned char *>(const_cast<char *>(data.data()));
```
Comment (jpeach) [inline lib/cppapi/GzipDeflateTransformation.cc]: `data()` isn't guaranteed to return a nul-terminated buffer.

File: lib/cppapi/GzipInflateTransformation.cc [hunk 2]
```diff
@@ -101,7 +100,7 @@ GzipInflateTransformation::consume(const string &data)
   vector<char> buffer(inflate_block_size);
 
   // Setup the compressed input
-  state_->z_stream_.next_in  = reinterpret_cast<unsigned char *>(const_cast<char *>(data.c_str()));
+  state_->z_stream_.next_in  = reinterpret_cast<unsigned char *>(const_cast<char *>(data.data()));
```
Comment (jpeach) [inline lib/cppapi/GzipInflateTransformation.cc]: `data()` isn't guaranteed to return a nul-terminated buffer.

File: plugins/experimental/webp_transform/ImageTransform.cc [hunk 3]
```diff
@@ -18,21 +18,20 @@
...
 
 #include <Magick++.h>
 
-using std::string;
 using namespace Magick;
 using namespace atscppapi;
 
-namespace
-{
+using std::string;
```
Comment (jpeach) [inline plugins/experimental/webp_transform/ImageTransform.cc]: Any reason to move this?

**Comment by ywkaras** [comment]
@jpeach I'm not sure what you want me to add to the commit message.  I think it's common knowledge that, other things being equal, code that doesn't use dynamic memory allocation is more efficient than code that does.

File: lib/cppapi/GzipDeflateTransformation.cc (hunk 1 above)
Comment (ywkaras) [reply lib/cppapi/GzipDeflateTransformation.cc]: Looking at the zlib documentation, I can't see any requirement that the sequence of bytes pointed to by next_in be nul terminated.

Comment (jpeach) [reply lib/cppapi/GzipDeflateTransformation.cc]: Ah I see, there's a `avail_in` member being set too. Makes sense.

**Comment by jpeach** [comment]
> @jpeach I'm not sure what you want me to add to the commit message. I think it's common
> knowledge that, other things being equal, code that doesn't use dynamic memory allocation is
> more efficient than code that does.

Don't make people read your mind, make it easy and spell it out. For example:

> Foo is creating a temporary copy of the data in bar in so that it can do whatever. Replacing this
> usage with `string_view` lets us remove the copy and pass the data directly to where it needs
> to go.

The commit message tells the reader the context that they otherwise have to construct from reading the code. It makes it easier to know what to look for in the commit and sets the reviewers expectations.

**Comment by ywkaras** [comment]
OK I changed the commit comment.

**Comment by jpeach** [comment]
thanks @ywkaras

**Pull request merged**